ktor-client/ktor-client-apache/jvm/src/io/ktor/client/engine/apache/ApacheHttpRequest.kt

```diff
@@ -30,39 +30,22 @@ internal suspend fun CloseableHttpAsyncClient.sendRequest(
             }
         }
         val result = HttpResponseData(status, requestTime, headers, version, body, callContext)
-        response.complete(result)
+        continuation.resume(result)
     }
 
     val callback = object : FutureCallback<Unit> {
         override fun failed(exception: Exception) {
             callContext.cancel()
-            response.completeExceptionally(exception)
+            continuation.cancel(exception)
         }
 
         override fun completed(result: Unit) {}
 
         override fun cancelled() {
             callContext.cancel()
-            response.cancel()
+            continuation.cancel()
```

Comment:
Is that true that only one of these handlers will be called and only once?

Reply:
Yes and we checked it early.